crates/breez-sdk/core/src/sdk.rs

```diff
@@ -292,74 +297,248 @@ impl BreezSdk {
+        let transfers = self
+            .spark_wallet
+            .list_transfers(None, Some(transfer_ids.clone()))
+            .await?;
+
+        for transfer in transfers {
+            let payment = Payment::try_from(transfer)?;
+            info!("Inserting previously pending bitcoin payment: {payment:?}");
+            self.storage.insert_payment(payment).await?;
+        }
+
+        Ok(())
+    }
+
+    async fn sync_pending_token_payments(
+        &self,
+        pending_token_payments: &[&Payment],
+    ) -> Result<(), SdkError> {
+        if pending_token_payments.is_empty() {
+            return Ok(());
+        }
+
+        let hash_pending_token_payments_map = pending_token_payments.iter().try_fold(
+            std::collections::HashMap::new(),
+            |mut acc: std::collections::HashMap<&_, Vec<_>>, payment| {
+                let details = payment
+                    .details
+                    .as_ref()
+                    .ok_or_else(|| SdkError::Generic("Payment details missing".to_string()))?;
+
+                if let PaymentDetails::Token { tx_hash, .. } = details {
+                    acc.entry(tx_hash).or_default().push(payment);
+                    Ok(acc)
+                } else {
+                    Err(SdkError::Generic(
+                        "Payment is not a token payment".to_string(),
+                    ))
+                }
+            },
+        )?;
+
+        // TODO: Deal with paging (is necessary if there are a lot of pending payments)
+        let token_transactions = self
+            .spark_wallet
+            .list_token_transactions(ListTokenTransactionsRequest {
+                token_transaction_hashes: hash_pending_token_payments_map
+                    .keys()
+                    .map(|k| (*k).to_string())
+                    .collect(),
+                ..Default::default()
+            })
+            .await?;
+
+        for token_transaction in token_transactions {
+            let is_transfer_transaction =
+                matches!(token_transaction.inputs, TokenInputs::Transfer(..));
+            let payment_status = PaymentStatus::from_token_transaction_status(
+                &token_transaction.status,
+                is_transfer_transaction,
+            );
+            if payment_status != PaymentStatus::Pending {
+                let payments_to_update = hash_pending_token_payments_map
+                    .get(&token_transaction.hash)
+                    .ok_or(SdkError::Generic("Payment not found".to_string()))?;
+                for payment in payments_to_update {
+                    // For now, updating the status is enough
+                    let mut updated_payment = (**payment).clone();
+                    updated_payment.status = payment_status;
+                    info!("Inserting previously pending token payment: {updated_payment:?}");
+                    self.storage.insert_payment(updated_payment).await?;
```

Comment:
It seems that if a pending payment was canceled (rolled back) it will stay forever in the db? Looks like we need a mechanism that will remove these from the storage? Perhaps API for set_pending_payments that will do that as part of the API?